File: iOS/KinKitchen/KinKitchen/Models/User/DietaryProfile.swift
//
//  DietaryProfile.swift
//  KinKitchen
//
//  Created by Greg Hudler on 8/30/26.
//

import Foundation

struct DietaryProfile: Codable, Identifiable {
    let id: UUID
    let userId: UUID
    var visibility: DietaryVisibility
    var notes: String?
    let createdAt: Date
    var updatedAt: Date

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case visibility
        case notes
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

enum DietaryVisibility: String, Codable, CaseIterable {
    case `private`
    case connections
    case gatherings
    case `public`
}

File: Android/app/src/main/java/com/pushtomaindev/kinkitchen/models/user/DietaryProfile.kt
package com.pushtomaindev.kinkitchen.models.user

import kotlinx.datetime.Instant
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Mirrors iOS `DietaryProfile`. */
@Serializable
data class DietaryProfile(
    val id: String,
    @SerialName("user_id") val userId: String,
    val visibility: DietaryVisibility,
    val notes: String? = null,
    @SerialName("created_at") val createdAt: Instant,
    @SerialName("updated_at") val updatedAt: Instant,
)

/** Mirrors iOS `DietaryVisibility`. Serialized values match the Swift raw values. */
@Serializable
enum class DietaryVisibility {
    @SerialName("private") PRIVATE,
    @SerialName("connections") CONNECTIONS,
    @SerialName("gatherings") GATHERINGS,
    @SerialName("public") PUBLIC;
}
